src/ParametersHandler/YarpImplementation/include/BipedalLocomotionControllers/ParametersHandler/YarpImplementation.h

```diff
@@ -55,60 +69,172 @@ class YarpImplementation : public IParametersHandler<YarpImplementation>
+     * @param object The object to copy
+     */
+    void set(const yarp::os::Searchable& searchable);
+
+    /**
+     * Get a parameter [int]
      * @param parameterName name of the parameter
      * @param parameter parameter
-     * @tparam T type of the parameter
      * @return true/false in case of success/failure
      */
-    template <typename T> bool getParameter(const std::string& parameterName, T& parameter) const;
+    bool getParameter(const std::string& parameterName, int& parameter) const final;
 
     /**
-     * Set a parameter in the handler.
+     * Get a parameter [double]
      * @param parameterName name of the parameter
      * @param parameter parameter
-     * @tparam T type of the parameter
+     * @return true/false in case of success/failure
      */
-    template <typename T> void setParameter(const std::string& parameterName, const T& parameter);
+    bool getParameter(const std::string& parameterName, double& parameter) const final;
 
     /**
-     * Set the handler from an object.
-     * @param object The object to copy
-     * @tparam T type of the object
+     * Get a parameter [std::string]
+     * @param parameterName name of the parameter
+     * @param parameter parameter
+     * @return true/false in case of success/failure
      */
-    void set(const yarp::os::Searchable& searchable);
+    bool getParameter(const std::string& parameterName, std::string& parameter) const final;
+
+    /**
+     * Get a parameter [bool]
+     * @param parameterName name of the parameter
+     * @param parameter parameter
+     * @return true/false in case of success/failure
+     */
+
+    bool getParameter(const std::string& parameterName, bool& parameter) const final;
+
+    /**
+     * Get a parameter [std::span<int>]
+     * @param parameterName name of the parameter
+     * @param parameter parameter
+     * @return true/false in case of success/failure
+     */
+    bool getParameter(const std::string& parameterName, const iDynTree::Span<int>& parameter) const final;
+
+    /**
+     * Get a parameter [std::span<double>]
+     * @param parameterName name of the parameter
+     * @param parameter parameter
+     * @return true/false in case of success/failure
+     */
+    bool getParameter(const std::string& parameterName, const iDynTree::Span<double>& parameter) const final;
+
+    /**
+     * Get a parameter [std::span<std::string>]
+     * @param parameterName name of the parameter
+     * @param parameter parameter
+     * @return true/false in case of success/failure
+     */
+    bool getParameter(const std::string& parameterName, const iDynTree::Span<std::string>& parameter) const final;
+
+    /**
+     * Get a parameter [std::vector<bool>]
+     * @param parameterName name of the parameter
+     * @param parameter parameter
+     * @return true/false in case of success/failure
+     */
+    bool getParameter(const std::string& parameterName, std::vector<bool>& parameter) const final;
```

Comment:
Why boolean and strings are handled differently? In both cases there is not an equivalent iDynTree or Eigen version. So why not using Span or vectors in both cases?

Reply:
Having a `span<bool>` may not work. I.e. `span<bool>` cannot be initialized from an  `std::vector<bool>`

Related issue: https://github.com/tcbrindle/span/issues/24

Reply:
Regarding the `std::string`, I implemented 
```cpp
setParameter(const std::string& parameterName, const string& parameter);
setParameter(const std::string& parameterName, const char* parameter);
```
because of [this](https://www.bfilipek.com/2019/07/surprising-conversions-char-bool.html)